src/Compilers/Core/Portable/PublicAPI.Unshipped.txt

```diff
@@ -203,6 +203,10 @@ Microsoft.CodeAnalysis.Operations.CommonConversion.IsNumeric.get -> bool
 Microsoft.CodeAnalysis.Operations.CommonConversion.IsReference.get -> bool
 Microsoft.CodeAnalysis.Operations.CommonConversion.IsUserDefined.get -> bool
 Microsoft.CodeAnalysis.Operations.CommonConversion.MethodSymbol.get -> Microsoft.CodeAnalysis.IMethodSymbol
+Microsoft.CodeAnalysis.Operations.ConditionalBranchKind
+Microsoft.CodeAnalysis.Operations.ConditionalBranchKind.IfFalse = 1 -> Microsoft.CodeAnalysis.Operations.ConditionalBranchKind
+Microsoft.CodeAnalysis.Operations.ConditionalBranchKind.IfNull = 2 -> Microsoft.CodeAnalysis.Operations.ConditionalBranchKind
```

Comment:
>CodeAnalysis [](start = 10, length = 12)

This feels out of place. Conditions should be boolean; we have existing expression nodes that can express a null test. Also it seems strange to have an IfNull but not an IfNotNull.

Reply:
> This feels out of place. Conditions should be boolean; we have existing expression nodes that can express a null test. Also it seems strange to have an IfNull but not an IfNotNull.

I see nothing wrong to have an easy way to represent a null check in the graph. Of course, we'll discuss that at the design meeting. But for now I am changing Condition to a Value and adding this new branch kind. If any follow-up is needed based on resolution of the meeting, I'll do them in a separate PR

Reply:
For now IfNotNull is not needed, but we can add it for completeness. Again let's have conversation at the design meeting.

Reply:
I cannot attend the design meeting due to recovering from surgery.

We have conditionals in many places in the languages C# and VB, and in none of these is there a primitive "null check" as a condition (as there is in C). Neither is there such a thing in either compiler's lowered form: we instead use a boolean expression which checks the input against null. This seems to be an arbitrarily specific combination of the equality and null primitives, and lowered forms should not be introducing such higher-level operations.